system/jlib/jstats.cpp

```diff
@@ -2108,6 +2169,51 @@ class CRootStatisticCollection : public CStatisticCollection
 };
 
 
+StatsScopeId globalScopeId(SSTglobal, (unsigned)0);
+class GlobalStatisticCollection : public CStatisticCollection
+{
+public:
+    GlobalStatisticCollection(IPropertyTree * root) : CStatisticCollection(nullptr, globalScopeId)
+    {
+        Owned<IPropertyTreeIterator> iter = root->getElements("*");
+        ForEach(*iter)
+        {
+            IPropertyTree * graphPT = &iter->query();
+
+            Owned<IPropertyTreeIterator> iter2 = graphPT->getElements("./*");
+            ForEach(*iter2)
+            {
+                IPropertyTree * sgPT = & iter2->query();
+                const char * sgName = sgPT->queryName();
+                if (strcmp(sgName, "node")==0)
+                    continue;
+                assertex(strncmp(sgName, "sg", 2)==0);
+                MemoryBuffer compressed;
+                sgPT->getPropBin("Stats", compressed);
+                if (!compressed.length())
+                    return;
+
+                MemoryBuffer serialized;
+                decompressToBuffer(serialized, compressed);
+                unsigned version;
+                serialized.read(version);
+                byte kind;
+                serialized.read(kind);
+
+                StatsScopeId id;
+                id.deserialize(serialized, version);
+                IStatisticCollection * collection = ensureSubScope(id, true);
+                collection->deserialize(serialized, version);
```

Comment:
Optimization for later.  Indicate to deserialize that it does not need to deserialize beyond a subgraph.  Would avoid all the activities being deserialized.
That option might be usable by other stats iterators.

Reply:
Noted.  I'll avoid deserialize unnecessary scopes.